router/handle.go

```diff
@@ -589,6 +595,45 @@ func (rt *Handle) findWorkerSlot(ctx context.Context, workers []*worker, job *jo
 	//#EndJobOrder
 }
 
+// checks if job is configured to drain or if it's retry limit is reached
+func (rt *Handle) drainOrRetryLimitReached(job *jobsdb.JobT) bool {
+	drain, _ := rt.drainer.Drain(job)
```

Comment:
[optional]  Would it make sense to get the drain reason and create a metric out of it?

Reply:
this is being done by the `worker`.
Changed things up a little so that this check is only done here and not again in the worker.
Consequently, changed the error message in case a job is drained as well: `routerutils.EnhanceJSON(job.LastJobStatus.ErrorResponse, "reason", abortReason)`